pkg/extensions/sync/on_demand.go

```diff
@@ -347,6 +350,13 @@ func syncRun(regCfg RegistryConfig, localRepo, remoteRepo, tag string, utils syn
 	}
 
 	err = pushSyncedLocalImage(localRepo, tag, localCachePath, utils.imageStore, log)
+	if errors.Is(err, zerr.ErrImageLintAnnotations) {
```

Comment:
no need to check this anymore, we have moved this check inside pushSyncedLocalImage()